src/Symfony/Component/Mime/Email.php

```diff
@@ -390,6 +390,10 @@ public function ensureValidity()
             throw new LogicException('A message must have a text or an HTML part or attachments.');
         }
 
+        if ('1' === $this->getHeaders()->getHeaderBody('X-Unsent')) {
```

Comment:
I added this here to avoid duplication but can override `ensureValidity` in `UnsentEmail` if that would be better.

Reply:
Is there a valid use-case for setting `X-Unsent: 1` in an instance of `Email`?

Otherwise having code in the parent class for the behavior of a child class sounds a bad design. The method `ensureValidity` can be overloaded in `DraftEmail` in order to remove wrong checks.

Reply:
This brings up a good point: what should we do if someone tries to send an email with mailer with `X-Unsent: 1`?

Reply:
I think we should throw an exception.